adapter/pkg/messaging/azure_connection.go

```diff
@@ -37,45 +50,181 @@ func init() {
+		if subManagerError != nil {
+			errorValue = errors.New("Error occurred while trying to get subscription manager from azure service bus for topic name : " +
+				key + ":" + subManagerError.Error())
+			return metaDataList, errorValue
+		}
+		logger.LoggerMgw.Info("[TEST][FEATURE_FLAG_REPLACE_EVENT_HUB] Subscription manager created for the " +
+			"topic " + key)
+		subscriptionMetaData.subscriptionManager = subManager
+		//We are creating a unique subscription for each adapter starts. Unused subscriptions will be deleted after
+		// idle for three days
+		uniqueID := uuid.New()
+
+		//In Azure service bus subscription names can contain letters, numbers, periods (.), hyphens (-), and
+		// underscores (_), up to 50 characters. Subscription names are also case-insensitive.
+		var subscriptionName = componentName + "_" + uniqueID.String() + "_sub"
+		var subscriptionCreationError error
+		func() {
+			ctx, cancel := context.WithCancel(parentContext)
+			defer cancel()
+			_, subscriptionCreationError = subManager.Put(ctx, subscriptionName, opts...)
+		}()
+		if subscriptionCreationError != nil {
+			errorValue = errors.New("Error occurred while trying to create subscription " + subscriptionName + "in azure service bus for topic name " +
+				key + ":" + subscriptionCreationError.Error())
+			return metaDataList, errorValue
+		}
+		logger.LoggerMgw.Info("[TEST][FEATURE_FLAG_REPLACE_EVENT_HUB] Subscription " + subscriptionName + " created")
+		subscriptionMetaData.subscriptionName = subscriptionName
+		subscriptionMetaData.topicName = key
+		metaDataList = append(metaDataList, subscriptionMetaData)
+	}
+	return metaDataList, nil
+}
+
+func getTopicList(topicManager *servicebus.TopicManager) ([]*servicebus.TopicEntity, error) {
+	var errorValue error
+	var getTopicListError error
+	var topicList []*servicebus.TopicEntity
+	func() {
+		ctx, cancel := context.WithCancel(context.Background())
+		defer cancel()
+		topicList, getTopicListError = topicManager.List(ctx)
+	}()
+	if getTopicListError != nil {
+		errorValue = errors.New("Error occurred while trying to get topic list from azure service bus:%v." + getTopicListError.Error())
+		return topicList, errorValue
+	}
+	logger.LoggerMgw.Info("[TEST][FEATURE_FLAG_REPLACE_EVENT_HUB] Topic list received ")
+	return topicList, nil
+}
+
+func isTopicExist(topicName string, availableTopicList []*servicebus.TopicEntity) bool {
+	for _, topic := range availableTopicList {
+		if strings.EqualFold(topic.Name, topicName) {
+			logger.LoggerMgw.Info("[TEST][FEATURE_FLAG_REPLACE_EVENT_HUB] Topic " + topicName + " Exist ")
+			return true
+		}
+	}
+	logger.LoggerMgw.Info("[TEST][FEATURE_FLAG_REPLACE_EVENT_HUB] Topic " + topicName + " does not Exist ")
+	return false
+}
+
+func logError(reconnectRetryCount int, reconnectInterval time.Duration, errVal error) {
+	retryAttemptMessage := ""
+	if reconnectRetryCount > 0 {
+		retryAttemptMessage = "Retry attempt : " + strconv.Itoa(reconnectRetryCount)
```

Comment:
Shall we include details like retry attempt to which endpoint

Reply:
This method gets the errVal which will have all the information regarding the error, what is the issue and which method caused it. There are multiple processes using the logError method and I wanted to keep it generic as errVal has the all the information of the issue . Does it make sense ? Shall we keep this as it is